crates/cxx-qt-lib/src/gui/qfontmetrics.rs

```diff
@@ -0,0 +1,146 @@
+        /// the entire string is used. The entire length of text is analysed even if len is substantially shorter.
+        #[rust_name = "horizontal_advance"]
+        fn horizontalAdvance(self: &QFontMetrics, text: &QString, len: i32) -> i32;
+
+        /// Returns the leading of the font.
+        fn leading(self: &QFontMetrics) -> i32;
+
+        /// Returns the distance from one base line to the next.
+        /// This value is always equal to leading()+height().
+        #[rust_name = "line_spacing"]
+        fn lineSpacing(self: &QFontMetrics) -> i32;
+
+        /// Returns the width of the underline and strikeout lines, adjusted for the point size of the font.
+        #[rust_name = "line_width"]
+        fn lineWidth(self: &QFontMetrics) -> i32;
+
+        /// Returns the width of the widest character in the font.
+        #[rust_name = "max_width"]
+        fn maxWidth(self: &QFontMetrics) -> i32;
+
+        /// Returns the minimum left bearing of the font.
+        #[rust_name = "min_left_bearing"]
+        fn minLeftBearing(self: &QFontMetrics) -> i32;
+
+        /// Returns the minimum right bearing of the font.
+        #[rust_name = "min_right_bearing"]
+        fn minRightBearing(self: &QFontMetrics) -> i32;
+
+        /// Returns the distance from the base line to where an overline should be drawn.
+        #[rust_name = "overline_position"]
+        fn overlinePos(self: &QFontMetrics) -> i32;
+
+        /// Returns the distance from the base line to where the strikeout line should be drawn.
+        #[rust_name = "strike_out_position"]
+        fn strikeOutPos(self: &QFontMetrics) -> i32;
+
+        /// Returns a tight bounding rectangle around the characters in the string specified by text.
+        /// The bounding rectangle always covers at least the set of pixels the text would cover if drawn at (0, 0).
+        #[rust_name = "tight_bounding_rect"]
+        fn tightBoundingRect(self: &QFontMetrics, text: &QString) -> QRect;
+
+        /// Returns the distance from the base line to where an underscore should be drawn.
+        #[rust_name = "underline_position"]
+        fn underlinePos(self: &QFontMetrics) -> i32;
+
+        /// Returns the 'x' height of the font. This is often but not always the same as the height of the character 'x'.
+        #[rust_name = "x_height"]
+        fn xHeight(self: &QFontMetrics) -> i32;
+    }
+
+    #[namespace = "rust::cxxqtlib1"]
+    unsafe extern "C++" {
+        include!("cxx-qt-lib/common.h");
+
+        #[doc(hidden)]
+        #[rust_name = "qfontmetrics_drop"]
+        fn drop(fontmetics: &mut QFontMetrics);
+
+        #[doc(hidden)]
+        #[rust_name = "qfontmetrics_clone"]
+        fn construct(font: &QFontMetrics) -> QFontMetrics;
+
+        #[doc(hidden)]
+        #[rust_name = "qfontmetrics_init_from_qfont"]
+        fn construct(font: &QFont) -> QFontMetrics;
+    }
+}
+
+#[repr(C)]
+pub struct QFontMetrics {
+    _cspec: MaybeUninit<i32>,
```

Comment:
I think it crashes because `MaybeUninit<i32>` is not of the size `std::size_t` (which is what you are asserting the size to be in C++.

This should instead be `MaybeUninit<usize>` for the size of the memory to be correct.